blog/content/post/2022-01-03-pcie_screamer.md

```diff
@@ -0,0 +1,218 @@
+
+## Vavado installation problem
+
+Instaling Vivado may be quite problematic. At first, there is a need to create
+an account at the Xilinx webpage and provide a lot of personal details like your
+name, company name, job function, and address.
+
+After downloading an installer and starting it, there is another problem.
+The software requires a lot of disk space. Selecting the least options possible
+requires 107.88 GB during installation. Complete installation may need even
+259.99 GB. Some of that space is freed after installation, but that doesn't
+change that you need a big enough HDD and free space.
+
+Installing that gargantuan pack of software takes a few hours too, but luckily
+it is a minuscule problem.
+
+## Script tested
+
+Simply having an IDE installed is not enough to program an FPGA. It also needs
+a correct design that can be programmed into the device. There are several
+projects that implement example designs for PCIeScreamer that can be later
+modified to meet our needs.
+
+For this usage, the
+[enjoy-digital/pcie_screamer](https://github.com/enjoy-digital/pcie_screamer)
+project was chosen. However, despite the simple instruction on the project
+[README](https://github.com/enjoy-digital/pcie_screamer) page, it is not very
+straightforward for first use.
+
+To prepare `enjoy-digital/pcie_screamer`, the following steps were tried:
+
+1. Install Vivado.
+1. Add Vivado to the `$PATH` environment variable.
+
+   ```
+   export PATH=$PATH:~/Xilinx/Vivado/2021.2/bin
+   ```
+
+   Your Vivado location may be different depending on options chosen during the
+   installation process and installed version.
+
+   You can check if you have done it correctly using the following command:
+
+   ```
+   vivado -version
+   ```
+
+1. Get and configure OpenCOD source code
+
+   ```
+   git clone https://github.com/openocd-org/openocd
+   cd openocd
+
+   git fetch https://review.openocd.org/openocd refs/changes/01/6801/1 && git checkout FETCH_HEAD
+
+   ./bootstrap
+   ./configure
+   ```
+
+1. Prepare a separate folder for files so the scripts won't mess inside your
+   important directory and change your current working directory to it.
+1. Clone
+   [enjoy-digital/pcie_screamer](https://github.com/enjoy-digital/pcie_screamer)
+   and checkout the correct branch
+
+   ```
+   git clone git@github.com:3mdeb/pcie_screamer.git
+   cd pcie_screamer
+   git checkout 01aee32f54f15163fac870cda3739ddd77b3c03c
+   ```
+
+2. Prepare [enjoy-digital/litex](https://github.com/enjoy-digital/litex)
+   environment
+
+   ```
+   wget https://raw.githubusercontent.com/enjoy-digital/litex/master/litex_setup.py
+   chmod +x litex_setup.py
+   ./litex_setup.py --init --install --user
+   ```
+
+   > This step will extract plenty of files in the directory, be sure to execute
+   > it from the `pcie_screamer` project directory.
+
+3. Now, the building of `Xilinx BIT data` is working
+
+   ```
+   ./pcie_screamer.py --build
+   ```
+
+4. However, the loading fails
```

Comment:
There is lack of information what type of hardware programer had been used. Was it JTAG programmer : "Xilinx Cable" of one of it's clones, or maybe one of serial programmers (USB2serial).